### ISSUE TITLE
Ability to singularize / pluralize included resource root key

### ISSUE THREAD

**Issue by thec0keman** [miragejs/miragejs#650]
Right now, [getHashForIncludedResource](https://github.com/miragejs/miragejs/blob/master/lib/serializer.js#L450) will call `keyForRelationship` with only the resource model name, so all included resources will be either plural or singular (depending on your serializer config).

However, it would be nice for `belongsTo` relationships to be able to return a singular root.   I think the only current alternative would be to scan the final payload in `serialize` for `belongsTo` keys and rename them accordingly.

One possible solution would be to pass the resource to `keyForRelationship`, though that would be an API change

**Comment by samselikoff** [comment]
Hm not quite tracking - could you give me an example of the payload you're trying to mock?

**Comment by thec0keman** [comment]
Yeah for sure.  I think mirage is optimized around the AMS pattern where a side-loaded `belongsTo` is serialized as an array;  So, for a post/author:
```
{
  post: Post,
  authors: Author[]
}
```

We have an API that will side-load belongsTo associations like:
```
{
  post: Post,
  author: Author
}
```

This obviously doesn't work for collections.

**Comment by samselikoff** [comment]
I see. Yes, AMS-inspired. Also because the root is the model name (not the relationship name) there could be other Authors that show up, based on the full inclusion taking place.

In your API how could you look at

```
{
  post: Post,
  author: Author
}
```

and know the author belongs to the post? Is it based on the URL being queried?

IOW, if you had a GET to `/posts/1` and included the posts author (belongsTo), as well as the posts comments and each comment's author, you'd need to end up with a list of authors in the final payload if you wanted that whole graph of data. Does your API handle such cases, and if so how does it disambiguate them?

**Comment by christianhegedues** [comment]
I have a similar question:
Is it possible to exclude the root element for a single response and map it only for collections?

```javascript
this.get('/users')
/**
 * Response:
 * 
 * {
 *  users: [
 *    { id: 1, name: 'a' },
 *    { id: 2, name: 'b' },
 *    { id: 3, name: 'c' },
 *    ...
 *  ]
 * }
 */

this.get('/users/:id')
/**
 * Response:
 * 
 * {
 *  id: 1,
 *  name: 'a' 
 * }
 */
```
